src/deluge/storage/flash_storage.cpp

```diff
@@ -367,7 +368,7 @@ void readSettings() {
 		defaultKeyMenu.lower = buffer[57];
 		defaultKeyMenu.upper = buffer[58];
 
-		defaultScale = buffer[59];
+		defaultScale = buffer[148];
```

Comment:
we should read the old location and leave a tombstone value (e.g. 0xff), using the new location if the old location contains the tombstone.

Reply:
Also, I'm assuming you're moving locations to have a larger datatype? But that isn't necessary for now so maybe finding a way to re-use the existing location would be ideal

Reply:
What about checking for firmware version? The thing is that RANDOM and NONE are adjacent to the existing scales so that doesn't allow us to insert more scales in the middle. Now that I moved Random and None to the end (254 and 255) to have free space for new scales, I need to use a different FlashStorage slot as if we use the existing one, and a user loads the official firmware on top of the community one, he will get rubbish from the presetScaleNotes array, pointing to bad memory places. I added different locations for 7, 6 and 5 note scales as then we can add more scales in the future, have them in order, and have the saved flashstorage be still valid when you upgrade firmware. I will add extra code to check for out of bounds indexes to reset it back to 0, in case the user downgrades to a previous community version with less scales.

Reply:
To avoid crashing the official firmware we could consider the selection "NONE" as a tombstone value for the old slot. So if the user goes back to official firmware they will have None as selected default scale, if they change it to something else (like Minor or Dorian or whatever), and then later install the community firmware again, we recover that value again to the new slot, and write None again in the old slot